registrations/serializers.py

```diff
@@ -164,3 +164,28 @@ class Meta:
 
     def get_expiration(self, obj):
         return obj.timestamp + timedelta(minutes=code_validity_duration(obj.seats))
+
+
+class MassEmailSerializer(serializers.Serializer):
+    subject = serializers.CharField()
+    body = serializers.CharField()
+    signups = serializers.ListField(
+        child=serializers.CharField(), allow_empty=False, required=False
+    )
```

Comment:
This is a list of IDs, no? Could use `PrimaryKeyRelatedField` instead: https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield

Or more concretely, something like this:
```python
    signups = serializers.PrimaryKeyRelatedField(
        queryset=SignUp.objects.all().exclude(email=None), many=True, required=False,
    )
```

With this, you can also get rid of `validate_signups(...)`. You'd also need to refactor your view a bit, replacing lines 285 to 294 with:
```python
signups = serializer.validated_data.get("signups", registration.signups.exclude(email=None))
```

Reply:
Changed as suggested. I also added MassEmailSignupsField class to have custom get_queryset method. Otherwise user could send emails to signups of other registrations
```
class MassEmailSignupsField(serializers.PrimaryKeyRelatedField):
    def get_queryset(self):
        registration = self.context["request"].data["registration"]

        return registration.signups.exclude(email=None)


class MassEmailSerializer(serializers.Serializer):
    subject = serializers.CharField()
    body = serializers.CharField()
    signups = MassEmailSignupsField(
        many=True,
        required=False,
    )
  ```

Reply:
Ah... very true. Good catch!